<a href="https://colab.research.google.com/github/IdrisJunaidAI/efficacy-ai-learning-hub/blob/main/MeanMedianModeExercise_IdrisJunaid_ITAI_1372.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Mean, Median, Mode and NumPy

This notebook demonstrates how to calculate mean, median, and mode for a cybersecurity dataset. The example uses failed login attempts collected during monitoring periods.

In [1]:
import numpy as np
from statistics import mode

## Cybersecurity Dataset

The values below represent the number of failed login attempts observed during ten monitoring periods.

In [2]:
failed_logins = np.array([4, 7, 5, 9, 7, 6, 7, 12, 5, 7])
print("Failed login attempts:", failed_logins)

Failed login attempts: [ 4  7  5  9  7  6  7 12  5  7]


## Calculate Mean, Median, and Mode

The mean provides the arithmetic average, the median identifies the middle value and is less affected by extreme values, and the mode identifies the value that occurs most frequently.

In [3]:
mean_value = np.mean(failed_logins)
median_value = np.median(failed_logins)
mode_value = mode(failed_logins)

print("Mean:", mean_value)
print("Median:", median_value)
print("Mode:", mode_value)

Mean: 6.9
Median: 7.0
Mode: 7


## Interpretation

The mean number of failed login attempts is 6.9. The median is 7.0, which represents the middle of the observed data. The mode is 7 because it occurs more frequently than any other value. In cybersecurity analytics, these measures can help establish normal activity, reduce the effect of unusual outliers, and identify frequently occurring security events.